### PR TITLE
docs(pie-monorepo): add descriptive jsdoc comments to WC props

### PR DESCRIPTION
**Pull request by jamieomaguire**
Add descriptive jsdoc comments to component props

### CODE REVIEW

File: packages/components/pie-modal/src/defs.ts [hunk 1]
```diff
@@ -1,8 +1,19 @@
 export const headingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
 
 export interface ModalProps {
+    /**
+     * The text to display in the modal's heading.
+     */
     heading: string;
+    /**
```
Comment (LTurns) [inline packages/components/pie-modal/src/defs.ts:8]: super nit but you're using full stops in modal but not in button or icon button 😄

Comment (ashleynolan) [reply packages/components/pie-modal/src/defs.ts:8]: @xander-marjoram would be proud

Comment (jamieomaguire) [reply packages/components/pie-modal/src/defs.ts:8]: Ugh. Removing them from here now

File: packages/components/pie-modal/src/defs.ts [hunk 2]
```diff
@@ -1,8 +1,19 @@
 export const headingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
 
 export interface ModalProps {
+    /**
+     * The text to display in the modal's heading.
```
Comment (ashleynolan) [inline packages/components/pie-modal/src/defs.ts:5]: Is it worth noting which of these are required and optional? (I think there's a `@required` directive if so).

If not required, I'd explicitly say here that the heading is optional.

Comment (jamieomaguire) [reply packages/components/pie-modal/src/defs.ts:5]: Conflicted about this. Seems super smelly to not leverage actual type systems to achieve this. Not sure how we go about it though. Can we look into that separately? I can't see a `@required` directive available

Comment (jamieomaguire) [reply packages/components/pie-modal/src/defs.ts:5]: Sounds good, but do you mind pointing me to an example of this in the wild? I honestly can't seem to see how we could achieve this!

**Review by raoufswe — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Pull request merged**